src/transformers/adapters/models/bert.py

```diff
@@ -15,162 +14,35 @@
     QuestionAnsweringHead,
     TaggingHead,
 )
-from ..layer import AdapterLayerBaseMixin
+from ..layer import AdapterLayer
 from ..model_mixin import InvertibleAdaptersMixin, ModelAdaptersMixin
 
 
 logger = logging.getLogger(__name__)
 
 
-class BertSelfOutputAdaptersMixin(AdapterLayerBaseMixin):
+# For backwards compatibility, BertSelfOutput inherits directly from AdapterLayer
+class BertSelfOutputAdaptersMixin(AdapterLayer):
```

Comment:
Why do some models still have AdapterMixins and others don't e.g. in the BART implementation, the `BartSelfAttentionAdaptersModule` is completely removed?

Reply:
Mainly for compatibility reasons. I didn't want to change the weight names and write yet another weight renaming method, so mixins are kept wherever necessary.

Reply:
However, we could go even further and also remove mixins such as `BartEncoderLayerAdaptersMixin` and just add the code to the main modeling classes as they mainly consist of creating child modules 🤔